src/modules/temperature_compensation/temperature_calibration/accel.cpp

```diff
@@ -167,6 +172,14 @@ int TemperatureCalibrationAccel::finish()
 
 int TemperatureCalibrationAccel::finish_sensor_instance(PerSensorData &data, int sensor_index)
 {
+	if (!data.has_valid_temperature) {
+		PX4_WARN("Result Accel %d does not have a valid temperature sensor", sensor_index);
+
+		uint32_t param = 0;
+		set_parameter("TC_A%d_ID", sensor_index, &param);
```

Comment:
This makes sense, but worries me slightly. We know about the cases that don't have temperature at all (now reporting NAN), but I'm not absolutely certain it's impossible for other sensors to never publish NAN (eg sporadic read error, etc).

What do you think?

Reply:
You are right, better to be sure on that point. I think I can reverse the logic : initialize with no temperature sensors and change that when a temperature measure is received. Calibration will exit immediately if the first temperature measure of all sensors is nan ( I hope it is not the case juste afetthe boot ...)

Reply:
> Calibration will exit immediately if the first temperature measure of all sensors is nan ( I hope it is not the case juste afetthe boot ...)

Briefly thinking about this I'd say it's unlikely, but also not impossible.

The structure of many of the newer IMU drivers where the temperature is read AFTER the FIFO. This doesn't apply to the sensors like the ICM20602 that have the temperature in the FIFO. https://github.com/PX4/Firmware/blob/7e5e8259f3bc644185569cb2292fd5ebd6bb532f/src/drivers/imu/invensense/icm20689/ICM20689.cpp#L265-L269

That means technically the very first reading with have a temperature of NAN. Now, these drivers start much earlier than the `temperature_compensation` module and run at a much higher priority, but some of them do have fairly long initial reset periods (30-100 ms).

Let's move forward with this change and keep an eye on it. I wouldn't be opposed to restructuring the drivers slightly so that we go out of our way to never publish NAN as a temperature unless there's no sensor or something is actually busted.